test/framework/hub.go

```diff
@@ -0,0 +1,95 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	clientcmd "k8s.io/client-go/tools/clientcmd"
+
+	ocmfeature "open-cluster-management.io/api/feature"
+
+	"open-cluster-management.io/ocm/pkg/operator/helpers"
+)
+
+// Hub represents a hub cluster, it holds:
+// * the clients to interact with the hub cluster
+// * the metadata of the hub
+// * the runtime data of the hub
+type Hub struct {
+	*OCMClients
+	ClusterManagerName      string
+	ClusterManagerNamespace string
+	ClusterCfg              *rest.Config
+}
+
+func NewHub(kubeconfig string) (*Hub, error) {
+	clusterCfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	clients, err := NewOCMClients(clusterCfg)
+	if err != nil {
+		return nil, err
+	}
+	return &Hub{
+		OCMClients: clients,
+		// the name of the ClusterManager object is constantly "cluster-manager" at the moment; The same name as deploy/cluster-manager/config/samples
+		ClusterManagerName:      "cluster-manager",
+		ClusterManagerNamespace: helpers.ClusterManagerDefaultNamespace,
+		ClusterCfg:              clusterCfg,
+	}, nil
+}
+
+func CheckHubReady(hub *Hub) error {
```

Comment:
could this be a func in hub struct rather than using hub as an input.

Reply:
also answered in https://github.com/open-cluster-management-io/ocm/pull/565#discussion_r1675316792